App Services Compression function randomly crashes.

# 🐞 Bug Report

### Affected Services [**REQUIRED**]
Anny App Service that using the Compression GZIP function 


### Is this a regression?
Fails in Docker and Kubernetes and local command-line. Seems to have been around since function was created, just not used much.

### Description and Minimal Reproduction [**REQUIRED**]
```
helm install edgex-levski -n edgex --set edgex.security.enabled=true --set edgex.storage.useHostPath=false --set edgex.storage.nonSharedVolumesClassName=ceph-block --set edgex.storage.sharedVolumesClassName=ceph-filesystem --set edgex.replicas.appservice.appsample=1 .
```

## 🔥 Exception or Error
<pre><code>
kubectl logs -f edgex-app-sample-5b867974cf-shcqp
Script for waiting on security bootstrapping ready-to-run
Thu Dec  1 21:06:03 UTC 2022 Executing waitFor with /app-service-configurable -cp=consul.http://edgex-core-consul:8500 --registry --confdir=/res waiting on tcp://edgex-security-bootstrapper:54329
level=INFO ts=2022-12-01T21:06:03.718433288Z app=security-bootstrapper source=config.go:392 msg="Loaded service configuration from /edgex-init/res/configuration.toml"
level=INFO ts=2022-12-01T21:06:03.719156422Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.KongDB.Port' by environment variable: STAGEGATE_KONGDB_PORT=5432"
level=INFO ts=2022-12-01T21:06:03.71919447Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.KongDB.ReadyPort' by environment variable: STAGEGATE_KONGDB_READYPORT=54325"
level=INFO ts=2022-12-01T21:06:03.719214324Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.SecretStoreSetup.Tokens.ReadyPort' by environment variable: STAGEGATE_SECRETSTORESETUP_TOKENS_READYPORT=54322"
level=INFO ts=2022-12-01T21:06:03.719225362Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.Registry.Host' by environment variable: STAGEGATE_REGISTRY_HOST=edgex-core-consul"
level=INFO ts=2022-12-01T21:06:03.719235282Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.KongDB.Host' by environment variable: STAGEGATE_KONGDB_HOST=edgex-kong-db"
level=INFO ts=2022-12-01T21:06:03.719244983Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.BootStrapper.StartPort' by environment variable: STAGEGATE_BOOTSTRAPPER_STARTPORT=54321"
level=INFO ts=2022-12-01T21:06:03.719253385Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.BootStrapper.Host' by environment variable: STAGEGATE_BOOTSTRAPPER_HOST=edgex-security-bootstrapper"
level=INFO ts=2022-12-01T21:06:03.719265768Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.Database.ReadyPort' by environment variable: STAGEGATE_DATABASE_READYPORT=6379"
level=INFO ts=2022-12-01T21:06:03.719274244Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.Database.Port' by environment variable: STAGEGATE_DATABASE_PORT=6379"
level=INFO ts=2022-12-01T21:06:03.719291117Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.Database.Host' by environment variable: STAGEGATE_DATABASE_HOST=edgex-redis"
level=INFO ts=2022-12-01T21:06:03.719303089Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.SecretStoreSetup.Host' by environment variable: STAGEGATE_SECRETSTORESETUP_HOST=edgex-security-secretstore-setup"
level=INFO ts=2022-12-01T21:06:03.719313876Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.Registry.Port' by environment variable: STAGEGATE_REGISTRY_PORT=8500"
level=INFO ts=2022-12-01T21:06:03.719325909Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'SecretStore.Port' by environment variable: SECRETSTORE_PORT=8200"
level=INFO ts=2022-12-01T21:06:03.719336372Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.WaitFor.Timeout' by environment variable: STAGEGATE_WAITFOR_TIMEOUT=60s"
level=INFO ts=2022-12-01T21:06:03.719347606Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.Ready.ToRunPort' by environment variable: STAGEGATE_READY_TORUNPORT=54329"
level=INFO ts=2022-12-01T21:06:03.719356766Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'SecretStore.Host' by environment variable: SECRETSTORE_HOST=edgex-vault"
level=INFO ts=2022-12-01T21:06:03.719386697Z app=security-bootstrapper source=variables.go:377 msg="Variables override of 'StageGate.Registry.ReadyPort' by environment variable: STAGEGATE_REGISTRY_READYPORT=54324"
level=INFO ts=2022-12-01T21:06:03.719530104Z app=security-bootstrapper source=config.go:556 msg="Using local configuration from file (17 envVars overrides applied)"
level=INFO ts=2022-12-01T21:06:03.719598992Z app=security-bootstrapper source=command.go:119 msg="Security bootstrapper running waitFor"
level=INFO ts=2022-12-01T21:06:03.719675887Z app=security-bootstrapper source=command.go:144 msg="Waiting for: [tcp://edgex-security-bootstrapper:54329] with timeout: [1m0s]"
level=INFO ts=2022-12-01T21:06:03.722306595Z app=security-bootstrapper source=command.go:214 msg="Connected to tcp://edgex-security-bootstrapper:54329"
Thu Dec  1 21:06:03 UTC 2022 Starting /app-service-configurable -cp=consul.http://edgex-core-consul:8500 --registry --confdir=/res ...
level=INFO ts=2022-12-01T21:06:03.736057835Z app=app-<profile> source=service.go:496 msg="Starting app-sample 2.3.0 "
level=INFO ts=2022-12-01T21:06:03.736262111Z app=app-sample source=variables.go:377 msg="Variables override of '-p/-profile' by environment variable: EDGEX_PROFILE=sample"
level=INFO ts=2022-12-01T21:06:03.738345059Z app=app-sample source=config.go:392 msg="Loaded service configuration from /res/sample/configuration.toml"
level=INFO ts=2022-12-01T21:06:03.73990077Z app=app-sample source=variables.go:377 msg="Variables override of 'Clients.core-metadata.Host' by environment variable: CLIENTS_CORE_METADATA_HOST=edgex-core-metadata"
level=INFO ts=2022-12-01T21:06:03.739955159Z app=app-sample source=variables.go:377 msg="Variables override of 'Clients.core-command.Host' by environment variable: CLIENTS_CORE_COMMAND_HOST=edgex-core-command"
level=INFO ts=2022-12-01T21:06:03.73996865Z app=app-sample source=variables.go:377 msg="Variables override of 'Service.Host' by environment variable: SERVICE_HOST=edgex-app-sample"
level=INFO ts=2022-12-01T21:06:03.739984881Z app=app-sample source=variables.go:377 msg="Variables override of 'Trigger.EdgexMessageBus.PublishHost.Host' by environment variable: TRIGGER_EDGEXMESSAGEBUS_PUBLISHHOST_HOST=edgex-redis"
level=INFO ts=2022-12-01T21:06:03.739998511Z app=app-sample source=variables.go:377 msg="Variables override of 'Clients.core-data.Host' by environment variable: CLIENTS_CORE_DATA_HOST=edgex-core-data"
level=INFO ts=2022-12-01T21:06:03.740025046Z app=app-sample source=variables.go:377 msg="Variables override of 'Registry.Host' by environment variable: REGISTRY_HOST=edgex-core-consul"
level=INFO ts=2022-12-01T21:06:03.740033004Z app=app-sample source=variables.go:377 msg="Variables override of 'Clients.support-notifications.Host' by environment variable: CLIENTS_SUPPORT_NOTIFICATIONS_HOST=edgex-support-notifications"
level=INFO ts=2022-12-01T21:06:03.740041363Z app=app-sample source=variables.go:377 msg="Variables override of 'SecretStore.Port' by environment variable: SECRETSTORE_PORT=8200"
level=INFO ts=2022-12-01T21:06:03.740053208Z app=app-sample source=variables.go:377 msg="Variables override of 'SecretStore.Host' by environment variable: SECRETSTORE_HOST=edgex-vault"
level=INFO ts=2022-12-01T21:06:03.740062839Z app=app-sample source=variables.go:377 msg="Variables override of 'Trigger.EdgexMessageBus.SubscribeHost.Host' by environment variable: TRIGGER_EDGEXMESSAGEBUS_SUBSCRIBEHOST_HOST=edgex-redis"
level=INFO ts=2022-12-01T21:06:03.74024577Z app=app-sample source=secret.go:63 msg="Creating SecretClient"
level=INFO ts=2022-12-01T21:06:03.740277859Z app=app-sample source=secret.go:70 msg="Reading secret store configuration and authentication token"
level=INFO ts=2022-12-01T21:06:03.740293569Z app=app-sample source=secret.go:177 msg="load token from file"
level=INFO ts=2022-12-01T21:06:03.742235525Z app=app-sample source=secret.go:88 msg="Attempting to create secret client"
level=INFO ts=2022-12-01T21:06:03.745114509Z app=app-sample source=secret.go:99 msg="Created SecretClient"
level=INFO ts=2022-12-01T21:06:03.745132845Z app=app-sample source=secret.go:104 msg="SecretsFile not set, skipping seeding of service secrets."
level=INFO ts=2022-12-01T21:06:03.74524798Z app=app-sample source=secrets.go:276 msg="kick off token renewal with interval: 30m0s"
level=INFO ts=2022-12-01T21:06:03.781516852Z app=app-sample source=secrets.go:148 msg="successfully generated Consul token for service app-sample"
level=INFO ts=2022-12-01T21:06:03.781559049Z app=app-sample source=config.go:161 msg="Using Configuration Provider access token of length 36"
level=INFO ts=2022-12-01T21:06:03.781572302Z app=app-sample source=config.go:367 msg="Using Configuration provider (consul) from: http://edgex-core-consul:8500 with base path of edgex/appservices/2.0/app-sample"
level=INFO ts=2022-12-01T21:06:03.792394416Z app=app-sample source=variables.go:377 msg="Variables override of 'Trigger.EdgexMessageBus.PublishHost.Host' by environment variable: TRIGGER_EDGEXMESSAGEBUS_PUBLISHHOST_HOST=edgex-redis"
level=INFO ts=2022-12-01T21:06:03.79242437Z app=app-sample source=variables.go:377 msg="Variables override of 'Clients.core-data.Host' by environment variable: CLIENTS_CORE_DATA_HOST=edgex-core-data"
level=INFO ts=2022-12-01T21:06:03.792432401Z app=app-sample source=variables.go:377 msg="Variables override of 'Clients.support-notifications.Host' by environment variable: CLIENTS_SUPPORT_NOTIFICATIONS_HOST=edgex-support-notifications"
level=INFO ts=2022-12-01T21:06:03.792440791Z app=app-sample source=variables.go:377 msg="Variables override of 'Registry.Host' by environment variable: REGISTRY_HOST=edgex-core-consul"
level=INFO ts=2022-12-01T21:06:03.79244687Z app=app-sample source=variables.go:377 msg="Variables override of 'SecretStore.Port' by environment variable: SECRETSTORE_PORT=8200"
level=INFO ts=2022-12-01T21:06:03.792457439Z app=app-sample source=variables.go:377 msg="Variables override of 'SecretStore.Host' by environment variable: SECRETSTORE_HOST=edgex-vault"
level=INFO ts=2022-12-01T21:06:03.79247199Z app=app-sample source=variables.go:377 msg="Variables override of 'Trigger.EdgexMessageBus.SubscribeHost.Host' by environment variable: TRIGGER_EDGEXMESSAGEBUS_SUBSCRIBEHOST_HOST=edgex-redis"
level=INFO ts=2022-12-01T21:06:03.792478653Z app=app-sample source=variables.go:377 msg="Variables override of 'Clients.core-metadata.Host' by environment variable: CLIENTS_CORE_METADATA_HOST=edgex-core-metadata"
level=INFO ts=2022-12-01T21:06:03.792486178Z app=app-sample source=variables.go:377 msg="Variables override of 'Clients.core-command.Host' by environment variable: CLIENTS_CORE_COMMAND_HOST=edgex-core-command"
level=INFO ts=2022-12-01T21:06:03.792491679Z app=app-sample source=variables.go:377 msg="Variables override of 'Service.Host' by environment variable: SERVICE_HOST=edgex-app-sample"
level=INFO ts=2022-12-01T21:06:03.792675178Z app=app-sample source=config.go:556 msg="Configuration has been pulled from Configuration provider (10 envVars overrides applied)"
level=INFO ts=2022-12-01T21:06:03.833324495Z app=app-sample source=secrets.go:148 msg="successfully generated Consul token for service app-sample"
level=INFO ts=2022-12-01T21:06:03.833366793Z app=app-sample source=registry.go:61 msg="Using Registry access token of length 36"
level=INFO ts=2022-12-01T21:06:03.833384033Z app=app-sample source=registry.go:85 msg="Using Registry (consul) from http://edgex-core-consul:8500"
level=INFO ts=2022-12-01T21:06:03.935319852Z app=app-sample source=clients.go:183 msg="Using registry for URL for 'core-command': http://edgex-core-command:59882"
level=INFO ts=2022-12-01T21:06:03.936039765Z app=app-sample source=clients.go:183 msg="Using registry for URL for 'core-data': http://edgex-core-data:59880"
level=INFO ts=2022-12-01T21:06:03.936549117Z app=app-sample source=clients.go:183 msg="Using registry for URL for 'core-metadata': http://edgex-core-metadata:59881"
level=INFO ts=2022-12-01T21:06:03.937021507Z app=app-sample source=clients.go:183 msg="Using registry for URL for 'support-notifications': http://edgex-support-notifications:59860"
level=INFO ts=2022-12-01T21:06:03.937156549Z app=app-sample source=telemetry.go:78 msg="Starting CPU Usage Average loop"
level=INFO ts=2022-12-01T21:06:03.941282006Z app=app-sample source=manager.go:122 msg="Metrics Manager started with a report interval of 30s"
level=INFO ts=2022-12-01T21:06:03.941314306Z app=app-sample source=bootstrap.go:238 msg="SecurityConsulTokensRequested metric registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941332384Z app=app-sample source=bootstrap.go:238 msg="SecurityConsulTokenDuration metric registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941337233Z app=app-sample source=bootstrap.go:238 msg="SecuritySecretsRequested metric registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941341155Z app=app-sample source=bootstrap.go:238 msg="SecuritySecretsStored metric registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941373203Z app=app-sample source=server.go:86 msg="Registering standard routes..."
level=INFO ts=2022-12-01T21:06:03.941400621Z app=app-sample source=configupdates.go:48 msg="Waiting for App Service configuration updates..."
level=INFO ts=2022-12-01T21:06:03.941483875Z app=app-sample source=service.go:559 msg="Service started in: 205.474712ms"
level=INFO ts=2022-12-01T21:06:03.941498276Z app=app-sample source=main.go:38 msg="Loading Configurable Pipeline..."
level=INFO ts=2022-12-01T21:06:03.941617132Z app=app-sample source=runtime.go:177 msg="PipelineMessagesProcessed-default-pipeline metric has been registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941630796Z app=app-sample source=runtime.go:177 msg="PipelineMessageProcessingTime-default-pipeline metric has been registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.94163954Z app=app-sample source=runtime.go:177 msg="PipelineProcessingErrors-default-pipeline metric has been registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941650205Z app=app-sample source=runtime.go:116 msg="Transforms set for `default-pipeline` pipeline"
level=INFO ts=2022-12-01T21:06:03.941662781Z app=app-sample source=runtime.go:177 msg="PipelineMessagesProcessed-float-pipeline metric has been registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941671393Z app=app-sample source=runtime.go:177 msg="PipelineMessageProcessingTime-float-pipeline metric has been registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941676151Z app=app-sample source=runtime.go:177 msg="PipelineProcessingErrors-float-pipeline metric has been registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941691974Z app=app-sample source=runtime.go:177 msg="PipelineMessagesProcessed-int8-16-pipeline metric has been registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941701952Z app=app-sample source=runtime.go:177 msg="PipelineMessageProcessingTime-int8-16-pipeline metric has been registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941707875Z app=app-sample source=runtime.go:177 msg="PipelineProcessingErrors-int8-16-pipeline metric has been registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941736472Z app=app-sample source=triggermessageprocessor.go:90 msg="MessagesReceived metric has been registered and will be reported"
level=INFO ts=2022-12-01T21:06:03.941748338Z app=app-sample source=triggermessageprocessor.go:96 msg="InvalidMessagesReceived metric has been registered and will be reported (if enabled)"
level=INFO ts=2022-12-01T21:06:03.941753096Z app=app-sample source=triggerfactory.go:51 msg="EdgeX MessageBus trigger selected"
level=INFO ts=2022-12-01T21:06:03.941760378Z app=app-sample source=messaging.go:66 msg="Initializing Message Bus Trigger for 'redis'"
level=INFO ts=2022-12-01T21:06:03.941778491Z app=app-sample source=messaging.go:290 msg="Setting options for secure MessageBus with AuthMode='usernamepassword' and SecretName='redisdb"
level=INFO ts=2022-12-01T21:06:03.941832552Z app=app-sample source=server.go:162 msg="Starting HTTP Web Server on address edgex-app-sample:59700"
level=INFO ts=2022-12-01T21:06:03.943101158Z app=app-sample source=messaging.go:104 msg="Subscribing to topic(s): 'edgex/events/#' @ redis://edgex-redis:6379"
level=INFO ts=2022-12-01T21:06:03.943120379Z app=app-sample source=messaging.go:113 msg="Publishing to topic: 'example' @ redis://edgex-redis:6379"
level=INFO ts=2022-12-01T21:06:03.9431397Z app=app-sample source=service.go:202 msg="StoreAndForward disabled. Not running retry loop."
level=INFO ts=2022-12-01T21:06:03.943145824Z app=app-sample source=service.go:205 msg="app-sample has Started"
level=INFO ts=2022-12-01T21:06:03.943205961Z app=app-sample source=messaging.go:125 msg="Waiting for messages from the MessageBus on the 'edgex/events/#' topic"
panic: runtime error: index out of range [586] with length 260

goroutine 81 [running]:
compress/flate.(*compressor).findMatch(0x563303aa9c08?, 0x563303264daa?, 0xc0ffffffff?, 0xc000542500?, 0xc000542740?)
	compress/flate/deflate.go:259 +0x2a9
compress/flate.(*compressor).deflate(0xc000822000)
	compress/flate/deflate.go:443 +0x215
compress/flate.(*compressor).close(0xc000822000)
	compress/flate/deflate.go:646 +0x43
compress/flate.(*Writer).Close(...)
	compress/flate/deflate.go:732
compress/gzip.(*Writer).Close(0xc0005ee210)
	compress/gzip/gzip.go:242 +0x87
github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/transforms.(*Compression).CompressWithGZIP(0xc0005b4c90, {0x563303ab6200, 0xc0006fa780}, {0x56330395c540, 0xc0005b4480})
	github.com/edgexfoundry/app-functions-sdk-go/v2@v2.3.0/pkg/transforms/compression.go:70 +0x31d
github.com/edgexfoundry/app-functions-sdk-go/v2/internal/runtime.(*GolangRuntime).ExecutePipeline(0xc0007281e0, {0x563303a56640, 0xc0006fa700}, 0xc0006fa780, 0xc000393de8, 0x0, 0x0)
	github.com/edgexfoundry/app-functions-sdk-go/v2@v2.3.0/internal/runtime/runtime.go:297 +0xf5
github.com/edgexfoundry/app-functions-sdk-go/v2/internal/runtime.(*GolangRuntime).ProcessMessage(0xc0007281e0, 0xc0006fa780, {0x563303a56640, 0xc0006fa700}, 0xc0006fa480)
	github.com/edgexfoundry/app-functions-sdk-go/v2@v2.3.0/internal/runtime/runtime.go:207 +0x40e
github.com/edgexfoundry/app-functions-sdk-go/v2/internal/app.(*triggerMessageProcessor).MessageReceived.func1(0xc0006fa480, 0xc00013c940, 0xc00000e090)
	github.com/edgexfoundry/app-functions-sdk-go/v2@v2.3.0/internal/app/triggermessageprocessor.go:185 +0x28c
created by github.com/edgexfoundry/app-functions-sdk-go/v2/internal/app.(*triggerMessageProcessor).MessageReceived
	github.com/edgexfoundry/app-functions-sdk-go/v2@v2.3.0/internal/app/triggermessageprocessor.go:171 +0x48c
</code></pre>


## 🌍  Your Environment
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"archive", BuildDate:"2021-07-22T00:00:00Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.8", GitCommit:"4a3b558c52eb6995b3c5c1db5e54111bd0645a64", GitTreeState:"clean", BuildDate:"2021-12-15T14:46:22Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}

Against "main" in edgex-examples helm chart

--- a/pkg/transforms/compression.go
+++ b/pkg/transforms/compression.go
@@ -55,6 +55,12 @@ func (compression *Compression) CompressWithGZIP(ctx interfaces.AppFunctionConte
 	var buf bytes.Buffer
 
 	gzipWriter := gzip.NewWriter(&buf)
+	defer func() {
+		defer func() {
+			// make sure writer is closed if error occurred before close is called below
+			_ = gzipWriter.Close()
+		}()
+	}()
 
 	_, err = gzipWriter.Write(rawData)
 	if err != nil {
@@ -87,6 +93,12 @@ func (compression *Compression) CompressWithZLIB(ctx interfaces.AppFunctionConte
 	var buf bytes.Buffer
 
 	zlibWriter := zlib.NewWriter(&buf)
+	defer func() {
+		defer func() {
+			// make sure writer is closed if error occurred before close is called below
+			_ = zlibWriter.Close()
+		}()
+	}()
 
 	_, err = zlibWriter.Write(byteData)
 	if err != nil {

--- a/pkg/transforms/compression.go
+++ b/pkg/transforms/compression.go
@@ -56,10 +56,8 @@ func (compression *Compression) CompressWithGZIP(ctx interfaces.AppFunctionConte
 
 	gzipWriter := gzip.NewWriter(&buf)
 	defer func() {
-		defer func() {
-			// make sure writer is closed if error occurred before close is called below
-			_ = gzipWriter.Close()
-		}()
+		// make sure writer is closed if error occurred before close is called below
+		_ = gzipWriter.Close()
 	}()
 
 	_, err = gzipWriter.Write(rawData)
@@ -94,10 +92,8 @@ func (compression *Compression) CompressWithZLIB(ctx interfaces.AppFunctionConte
 
 	zlibWriter := zlib.NewWriter(&buf)
 	defer func() {
-		defer func() {
-			// make sure writer is closed if error occurred before close is called below
-			_ = zlibWriter.Close()
-		}()
+		// make sure writer is closed if error occurred before close is called below
+		_ = zlibWriter.Close()
 	}()
 
 	_, err = zlibWriter.Write(byteData)

--- a/pkg/transforms/compression.go
+++ b/pkg/transforms/compression.go
@@ -33,8 +33,6 @@ import (
 )
 
 type Compression struct {
-	gzipWriter *gzip.Writer
-	zlibWriter *zlib.Writer
 }
 
 // NewCompression creates, initializes and returns a new instance of Compression
@@ -56,18 +54,14 @@ func (compression *Compression) CompressWithGZIP(ctx interfaces.AppFunctionConte
 	}
 	var buf bytes.Buffer
 
-	if compression.gzipWriter == nil {
-		compression.gzipWriter = gzip.NewWriter(&buf)
-	} else {
-		compression.gzipWriter.Reset(&buf)
-	}
+	gzipWriter := gzip.NewWriter(&buf)
 
-	_, err = compression.gzipWriter.Write(rawData)
+	_, err = gzipWriter.Write(rawData)
 	if err != nil {
 		return false, fmt.Errorf("unable to write GZIP data in pipeline '%s': %s", ctx.PipelineId(), err.Error())
 	}
 
-	err = compression.gzipWriter.Close()
+	err = gzipWriter.Close()
 	if err != nil {
 		return false, fmt.Errorf("unable to close GZIP data in pipeline '%s': %s", ctx.PipelineId(), err.Error())
 	}
@@ -76,7 +70,6 @@ func (compression *Compression) CompressWithGZIP(ctx interfaces.AppFunctionConte
 	ctx.SetResponseContentType(common.ContentTypeText)
 
 	return true, bytesBufferToBase64(ctx.LoggingClient(), buf)
-
 }
 
 // CompressWithZLIB compresses data received as either a string,[]byte, or json.Marshaller using zlib algorithm
@@ -93,18 +86,14 @@ func (compression *Compression) CompressWithZLIB(ctx interfaces.AppFunctionConte
 	}
 	var buf bytes.Buffer
 
-	if compression.zlibWriter == nil {
-		compression.zlibWriter = zlib.NewWriter(&buf)
-	} else {
-		compression.zlibWriter.Reset(&buf)
-	}
+	zlibWriter := zlib.NewWriter(&buf)
 
-	_, err = compression.zlibWriter.Write(byteData)
+	_, err = zlibWriter.Write(byteData)
 	if err != nil {
 		return false, fmt.Errorf("unable to write ZLIB data in pipeline '%s': %s", ctx.PipelineId(), err.Error())
 	}
 
-	err = compression.zlibWriter.Close()
+	err = zlibWriter.Close()
 	if err != nil {
 		return false, fmt.Errorf("unable to close ZLIB data in pipeline '%s': %s", ctx.PipelineId(), err.Error())
 	}
@@ -113,7 +102,6 @@ func (compression *Compression) CompressWithZLIB(ctx interfaces.AppFunctionConte
 	ctx.SetResponseContentType(common.ContentTypeText)
 
 	return true, bytesBufferToBase64(ctx.LoggingClient(), buf)
-
 }
 
 func bytesBufferToBase64(lc logger.LoggingClient, buf bytes.Buffer) []byte {

